Ensure RemoteImageDecoderAVFProxy messages are not sent when MediaPlaybackEnabled is false

szewai opened the pull request:
#### a317744f5e1459934067237e26c59dea9c0e531b
<pre>
Ensure RemoteImageDecoderAVFProxy messages are not sent when MediaPlaybackEnabled is false
<a href="https://bugs.webkit.org/show_bug.cgi?id=284935">https://bugs.webkit.org/show_bug.cgi?id=284935</a>
<a href="https://rdar.apple.com/141731913">rdar://141731913</a>

Reviewed by Youenn Fablet.

When MediaPlaybackEnabled flag is false, it means media playback functionalities should be disabled, and web process is
not expected to send related messages to UI process or GPU process (they will treat related message as invalid message).
To ensure web process does not send RemoteImageDecoderAVFProxy messages, this patch makes RemoteImageDecoderAVFManager
does not create RemoteImageDecoderAVF when MediaPlaybackEnabled is false.

* Source/WebKit/WebProcess/GPU/media/RemoteImageDecoderAVFManager.cpp:
(WebKit::RemoteImageDecoderAVFManager::createImageDecoder):

Canonical link: <a href="https://commits.webkit.org/288207@main">https://commits.webkit.org/288207@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/fd510121f32d71a6b56a6df5c2e3587e10e54c53

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/82120 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/1655 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/36087 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/86695 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/33158 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/1698 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/9482 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/64007 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/21739 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/85190 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/1263 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/74713 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/44288 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/1162 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/31586 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/72459 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/29505 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/88127 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/9368 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/6685 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/72387 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/9557 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/70530 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/71607 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/17862 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/15735 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/14640 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/777 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/9324 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/14855 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/9170 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/12688 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/10972 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

youennf commented on Source/WebKit/WebProcess/WebProcess.cpp:
```diff
@@ -2454,6 +2454,10 @@ bool WebProcess::requiresScriptTelemetryForURL(const URL& url, const WebCore::Se
 void WebProcess::enableMediaPlayback()
 {
+#if ENABLE(GPU_PROCESS) && HAVE(AVASSETREADER)
+    protectedRemoteImageDecoderAVFManager()->enableMediaPlayback();
```
We could make WebProcess::m_remoteImageDecoderAVFManager const so that we would not have to use protectedRemoteImageDecoderAVFManager at all.

szewai committed:
Ensure RemoteImageDecoderAVFProxy messages are not sent when MediaPlaybackEnabled is false
https://bugs.webkit.org/show_bug.cgi?id=284935
rdar://141731913

Reviewed by Youenn Fablet.

When MediaPlaybackEnabled flag is false, it means media playback functionalities should be disabled, and web process is
not expected to send related messages to UI process or GPU process (they will treat related message as invalid message).
To ensure web process does not send RemoteImageDecoderAVFProxy messages, this patch makes RemoteImageDecoderAVFManager
does not create RemoteImageDecoderAVF when MediaPlaybackEnabled is false.

* Source/WebKit/WebProcess/GPU/media/RemoteImageDecoderAVFManager.cpp:
(WebKit::RemoteImageDecoderAVFManager::createImageDecoder):

Canonical link: https://commits.webkit.org/288207@main
